### DISCUSSION TITLE
Python based ldap sync

### DISCUSSION THREAD

**Discussion by sirtoobii** [dani-garcia/vaultwarden#3582] [Show and tell]
Hello everyone

We use Vaultwarden in a medium sized environment (~80 people) with high turnover of users joining and leaving our group. We have a company-wide LDAP server that reflects these changes very closely. Therefore, the integration of this LDAP directory seemed to make sense.

However, both existing solutions have a disadvantage:
- [Directory Connector](https://bitwarden.com/help/directory-sync): Does not invite users
- [vaultwarden_ldap](https://github.com/ViViDboarder/vaultwarden_ldap): Only invites users, but does not disable users who leave the group

To mitigate both of these shortcomings, I present you the **[vaultwarden_ldap_sync](https://github.com/sirtoobii/vaultwarden_ldap_sync)** project. Some of the highlights:
- Automatically invites users who are present on LDAP
- Disables users who disappeared from LDAP
- Users can be "untied" from automatic management when an admin manually activates a user after they have been automatically deactivated
- We even survive user-initiated email changes
- Fully dockerized setup

I look forward to your feedback and ideas :)

**Comment by BlackDex** [comment]
Please try the `:testing` tagged container images.
These are now compatible with the Directory Connector v2022.11.0 and should invite all the users and also supports groups sync.
